app/controllers/api/avatars.php

```diff
@@ -635,6 +637,243 @@
+                'height' => $browserHeight
+            ]
+        ];
+
+        // Add scale if not default
+        if ($scale != 1) {
+            $config['scale'] = $scale;
+        }
+
+        // Add fullPage to viewport if enabled
+        if ($fullpage) {
+            $config['viewport']['fullPage'] = true;
+        }
+
+        // Add optional parameters only if they have meaningful values
+        if (!empty($userAgent)) {
+            $config['userAgent'] = $userAgent;
+        }
+
+        if ($fullpage) {
+            $config['fullPage'] = true;
+        }
+
+        if (!empty($locale)) {
+            $config['locale'] = $locale;
+        }
+
+        if (!empty($timezone)) {
+            $config['timezoneId'] = $timezone;
+        }
+
+        // Add geolocation if any coordinates are provided
+        if ($latitude != 0 || $longitude != 0) {
+            $config['geolocation'] = [
+                'latitude' => $latitude,
+                'longitude' => $longitude,
+                'accuracy' => $accuracy
+            ];
+        }
+
+        if ($touch) {
+            $config['hasTouch'] = true;
+        }
+
+        // Add permissions if provided
+        if (!empty($permissions)) {
+            $config['permissions'] = $permissions;
+        }
+
+        // Manually handle the config to ensure headers is an object but arrays remain arrays
+        $finalConfig = [
+            'url' => $config['url'],
+            'theme' => $config['theme'],
+            'headers' => $config['headers'], // Keep as object
+            'sleep' => $config['sleep'],
+            'viewport' => $config['viewport'] // Keep as object
+        ];
+        
+        // Add scale if not default
+        if ($scale != 1) {
+            $finalConfig['scale'] = $scale;
+        }
+
+        // Add optional parameters that were set, preserving arrays as arrays
+        if (!empty($userAgent)) {
+            $finalConfig['userAgent'] = $userAgent;
+        }
+
+        if ($fullpage) {
+            $finalConfig['fullPage'] = true;
+        }
+
+        if (!empty($locale)) {
+            $finalConfig['locale'] = $locale;
+        }
+
+        if (!empty($timezone)) {
+            $finalConfig['timezoneId'] = $timezone;
+        }
+
+        // Add geolocation if any coordinates are provided
+        if ($latitude != 0 || $longitude != 0) {
+            $finalConfig['geolocation'] = [
+                'latitude' => $latitude,
+                'longitude' => $longitude,
+                'accuracy' => $accuracy
+            ];
+        }
+
+        if ($touch) {
+            $finalConfig['hasTouch'] = true;
+        }
+
+        // Add permissions if provided (preserve as array)
+        if (!empty($permissions)) {
+            $finalConfig['permissions'] = $permissions; // Keep as array
+        }
+
+        $config = $finalConfig;
+
+        try {
+            $browserEndpoint = Config::getParam('_APP_BROWSER_HOST', 'http://appwrite-browser:3000/v1');
```

Comment:
I think this should be System::getEnv

Reply:
Fixed. Was copy-pasted from builds worker - fixed there too.